src/sage/groups/matrix_gps/linear.py

```diff
@@ -307,3 +307,53 @@ def _check_matrix(self, x, *args):
         else:
             if x.determinant() == 0:
                 raise TypeError('matrix must non-zero determinant')
+
+    def order(self):
+        """
+        Return the order of ``self``.
+
+        EXAMPLES::
+
+            sage: G = SL(3, GF(5))
+            sage: G.order()
+            372000
+
+        TESTS:
+
+        Check if :trac:`36876` is fixed::
+            sage: SL(1, QQ).order()
+            1
+            sage: SL(1, ZZ).cardinality()
+            1
+
+        Check if :trac:`35490` is fixed::
+            sage: q = 7
+            sage: FqT.<T> = GF(q)[]
+            sage: N = T^2+1
+            sage: FqTN = QuotientRing(FqT, N*FqT)
+            sage: S = SL(2,FqTN)
+            sage: S.is_finite()
+            True
+            sage: S.order()
+            117600
+        """
+        from functools import reduce
+        import operator
+        from sage.rings.infinity import Infinity
+
+        n = self.degree()
+
+        if self.base_ring().is_finite():
+            q = self.base_ring().order()
+            if self._special:
+                return reduce(operator.mul, (q**n - q**i for i in range(0, n)), 1) / (q-1)
+            return reduce(operator.mul, (q**n - q**i for i in range(0, n)), 1)
```

Comment:
In 349 and 350 you can use `prod`, that should be more readable.  Also, `range(0, n)` is the same as `range(n)`.

It might also make sense to first compute `prod(q**n - q**i for i in range(n))`, and only divide by `q-1` if `self._special`.